rasa/core/policies/ted_policy.py

```diff
@@ -0,0 +1,642 @@
+
+
+class TEDPolicy(Policy):
+    """Transformer Embedding Dialogue (TED) Policy.
+
+    The policy used in our paper https://arxiv.org/abs/1910.00486.
+    """
+
+    SUPPORTS_ONLINE_TRAINING = True
+
+    # default properties (DOC MARKER - don't remove)
+    defaults = {
+        # nn architecture
+        # a list of hidden layers sizes before dialogue and action embedding layers
+        # number of hidden layers is equal to the length of this list
+        HIDDEN_LAYERS_SIZES: {DIALOGUE: [], LABEL: []},
+        # number of units in transformer
+        TRANSFORMER_SIZE: 128,
+        # number of transformer layers
+        NUM_TRANSFORMER_LAYERS: 1,
+        # max sequence length
+        MAX_SEQUENCE_LENGTH: 256,
+        # number of attention heads in transformer
+        NUM_HEADS: 4,
+        # if true use key relative embeddings in attention
+        KEY_RELATIVE_ATTENTION: False,
+        # if true use key relative embeddings in attention
+        VALUE_RELATIVE_ATTENTION: False,
+        # max position for relative embeddings
+        MAX_RELATIVE_POSITION: None,
+        # training parameters
+        # initial and final batch sizes:
+        # batch size will be linearly increased for each epoch
+        BATCH_SIZES: [8, 32],
+        # how to create batches
+        BATCH_STRATEGY: "balanced",  # 'sequence' or 'balanced'
+        # number of epochs
+        EPOCHS: 1,
+        # set random seed to any int to get reproducible results
+        RANDOM_SEED: None,
+        # embedding parameters
+        # dimension size of embedding vectors
+        EMBEDDING_DIMENSION: 20,
+        # the type of the similarity
+        NUM_NEG: 20,
+        # flag if minimize only maximum similarity over incorrect labels
+        SIMILARITY_TYPE: "auto",  # 'auto' or 'cosine' or 'inner'
+        # the type of the loss function
+        LOSS_TYPE: "softmax",  # 'softmax' or 'margin'
+        # number of top actions to normalize scores for softmax loss_type
+        # set to 0 to turn off normalization
+        RANKING_LENGTH: 10,
+        # how similar the algorithm should try
+        # to make embedding vectors for correct labels
+        MAX_POS_SIM: 0.8,  # should be 0.0 < ... < 1.0 for 'cosine'
+        # maximum negative similarity for incorrect labels
+        MAX_NEG_SIM: -0.2,  # should be -1.0 < ... < 1.0 for 'cosine'
+        # the number of incorrect labels, the algorithm will minimize
+        # their similarity to the user input during training
+        USE_MAX_NEG_SIM: True,  # flag which loss function to use
+        # scale loss inverse proportionally to confidence of correct prediction
+        SCALE_LOSS: True,
+        # regularization
+        # the scale of regularization
+        REGULARIZATION_CONSTANT: 0.001,
+        # the scale of how important is to minimize the maximum similarity
+        # between embeddings of different labels
+        NEGATIVE_MARGIN_SCALE: 0.8,
+        # dropout rate for dial nn
```

Comment:
please spell this out. `nn` is a common acronym, but `dial` for sure isn't
```suggestion
        # dropout rate for dialogue nn
```